backend/src/model/entity.rs

```diff
@@ -11,70 +11,230 @@ use diesel::{Identifiable, Insertable, Queryable};
-    pub cultivation: Option<String>,
-    pub environment: Option<String>,
+    ///
     pub material_uses: Option<String>,
+    ///
     pub functions: Option<String>,
-    pub provides_forage_for: Option<String>,
-    pub provides_shelter_for: Option<String>,
-    pub hardiness_zone: Option<i16>,
+    ///
     pub heat_zone: Option<i16>,
-    pub water: Option<Water>,
-    pub sun: Option<Sun>,
+    ///
     pub shade: Option<String>,
+    /// The list of enum values.
     pub soil_ph: Option<Vec<Option<SoilPh>>>,
+    /// The list of enum values.
     pub soil_texture: Option<Vec<Option<SoilTexture>>>,
+    /// The list of enum values.
     pub soil_water_retention: Option<Vec<Option<SoilWaterRetention>>>,
+    ///
     pub environmental_tolerances: Option<Vec<Option<String>>>,
-    pub native_climate_zones: Option<String>,
-    pub adapted_climate_zones: Option<String>,
+    ///
     pub native_geographical_range: Option<String>,
+    ///
     pub native_environment: Option<String>,
+    ///
     pub ecosystem_niche: Option<String>,
-    pub root_zone_tendancy: Option<RootZoneTendancy>,
+    /// Enum value.
     pub deciduous_or_evergreen: Option<DeciduousOrEvergreen>,
+    /// Enum value.
     pub herbaceous_or_woody: Option<HerbaceousOrWoody>,
+    /// The list of enum values.
     pub life_cycle: Option<Vec<Option<LifeCycle>>>,
-    pub growth_rate: Option<GrowthRate>,
-    pub mature_size_height: Option<String>,
-    pub mature_size_width: Option<String>,
+    /// The list of enum values.
+    pub growth_rate: Option<Vec<Option<GrowthRate>>>,
+    ///
+    pub height: Option<String>,
+    ///
+    pub width: Option<String>,
+    /// The list of enum values.
     pub fertility: Option<Vec<Option<Fertility>>>,
+    ///
     pub pollinators: Option<String>,
+    ///
     pub flower_colour: Option<String>,
+    /// Enum value.
     pub flower_type: Option<FlowerType>,
+    /// The creation date of the entry.
     pub created_at: NaiveDateTime,
+    /// The last update date of the entry.
     pub updated_at: NaiveDateTime,
+    ///
     pub has_drought_tolerance: Option<bool>,
+    ///
     pub tolerates_wind: Option<bool>,
+    /// The list of the references of the plant.
     pub plant_references: Option<Vec<Option<String>>>,
+    /// Boolean value indicating whether the plant is a tree. The initial value is to `True` if  herbaceous_or_woody (woody) and life_cycle (perennial)
     pub is_tree: Option<bool>,
+    /// Enum value. The initial value is to `light feeder` if "Nutritionally poor soil" in `environmental_tolerances` is present.
     pub nutrition_demand: Option<NutritionDemand>,
+    /// Number value between -1..6 (-1 should be printed as 00)
     pub preferable_permaculture_zone: Option<i16>,
+    /// Date value fetched from PracticalPlants page showing the last modification date of the plant.
     pub article_last_modified_at: Option<NaiveDateTime>,
+    ///
+    pub hardiness_zone: Option<String>,
```

Comment:
This should be an int.

Reply:
These values are coming as partial ranges and partial numbers from the datasets, so we first implemented them as ranges, which worked pretty well on the PSQL side. However, there were some complications on the backend side. I was trying to figure out how the ranges should be imported in the backend for quite a while, but all my attempts were unsuccessful. The main problem is that diesel doesn't know how to interpret the ranges coming from the PSQL without telling him how to parse such values in `FromSQL` and `ToSQL` constructions. So I thought that sanitized and clean entries are currently more valuable for the project and could be adapted by backend experts in the future